src/libraries/Pool.sol

```diff
@@ -427,7 +427,10 @@ library Pool {
                 }
             } else if (state.sqrtPriceX96 != step.sqrtPriceStartX96) {
                 // recompute unless we're on a lower tick boundary (i.e. already transitioned ticks), and haven't moved
-                state.tick = TickMath.getTickAtSqrtPrice(state.sqrtPriceX96);
+                if (!TickMath.isSqrtPriceInTick(state.sqrtPriceX96, state.tick, step.tickNext, zeroForOne)) {
+                    // recomputing does not occur if we obviously did not change the tick
+                    state.tick = TickMath.getTickAtSqrtPrice(state.sqrtPriceX96);
+                }
```

Comment:
It's a nice arbitrage between `getSqrtPriceAtTick` and `getTickAtSqrtPrice`. But in order for the protocol team to adopt it, it could use more explanations. Because this part of code is where KyberSwap messed up if I remember correctly.

Reply:
It could use formal verification like:

For the case `zeroForOne == true`, `state.sqrtPriceX96 >= step.sqrtPriceNextX96`. So if `state.tick == step.tickNext`, `state.sqrtPriceX96 >= getSqrtPriceAtTick(state.tick)`. Because `zeroForOne == true`, `getTickAtSqrtPrice(state.sqrtPriceX96) <= state.tick`. So if `state.sqrtPriceX96 >= getSqrtPriceAtTick(state.tick)`, `state.tick` remains the same.

Still, the above proof is sloppy.